internal/runbits/runtime/refresh.go

```diff
@@ -0,0 +1,113 @@
+package runtime
+
+import (
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/rtutils"
+	"github.com/ActiveState/cli/internal/rtutils/ptr"
+	"github.com/ActiveState/cli/internal/runbits"
+	"github.com/ActiveState/cli/internal/runbits/buildscript"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+// RefreshRuntime should be called after runtime mutations.
+func RefreshRuntime(
+	auth *authentication.Auth,
+	out output.Outputer,
+	an analytics.Dispatcher,
+	proj *project.Project,
+	commitID strfmt.UUID,
+	changed bool,
+	trigger target.Trigger,
+	svcm *model.SvcModel,
+	cfg Configurable,
+) (rerr error) {
+	target := target.NewProjectTarget(proj, &commitID, trigger)
+	rt, err := runtime.New(target, an, svcm, auth, cfg, out)
+	if err != nil {
+		return locale.WrapError(err, "err_packages_update_runtime_init", "Could not initialize runtime.")
+	}
+
+	if !changed && !rt.NeedsUpdate() {
+		out.Notice(locale.Tl("pkg_already_uptodate", "Requested dependencies are already configured and installed."))
+		return nil
+	}
+
+	if rt.NeedsUpdate() {
+		if !rt.HasCache() {
+			out.Notice(output.Title(locale.T("install_runtime")))
+			out.Notice(locale.T("install_runtime_info"))
+		} else {
+			out.Notice(output.Title(locale.T("update_runtime")))
+			out.Notice(locale.T("update_runtime_info"))
+		}
+	}
+
+	return RefreshRuntimeByReference(rt, auth, out, proj, cfg)
+}
+
+// RefreshRuntimeByReference will update the given runtime if necessary. Unlike RefreshRuntime this won't print any UI
+// except for the progress of sourcing the runtime.
+func RefreshRuntimeByReference(
```

Comment:
From the docstring and the function logic it's not clear why this has `ByReference` in the name. I could be missing something though.